### PR TITLE
Feature/rc to dep

### PR DESCRIPTION
**Pull request by scruplelesswizard**
Swap out Replication Controllers for Deployments for the cluster objects

### CODE REVIEW

**Comment by scruplelesswizard** [comment]
@scheeles @sttts can you please review this? I squashed the commits into one FYI

**Comment by sttts** [comment]
The commits do much more than the PR description.

File: controller/cluster/pending.go [hunk 1]
```diff
@@ -377,66 +373,81 @@ func (cc *clusterController) launchingCheckTLSSecret(c *api.Cluster) error {
...
+		if err != nil {
+			return err
+		}
+
+		cc.recordClusterEvent(c, "launching", "Created ingress")
+	}
 	return nil
 }
 
 func (cc *clusterController) launchingCheckReplicationController(c *api.Cluster) error {
```
Comment (sttts) [inline controller/cluster/pending.go]: Rename RC -> Deployment

**Comment by scruplelesswizard** [comment]
Sorry, the PR was targeting the wrong branch

File: controller/cluster/pending.go (hunk 1 above)
Comment (scruplelesswizard) [reply controller/cluster/pending.go]: Resolved

**Comment by scruplelesswizard** [comment]
@sttts @scheeles can you please re-review?

Please merge with kubermatic/kubermatic#62

**Pull request merged**